csrc/fusion.cpp

```diff
@@ -797,28 +797,37 @@ void Fusion::aliasOutputToInput(
     }
   }
 
-  NVF_ERROR(
-      isAliasCompatible(input, output),
-      "The input and output values are not alias-compatible.");
-  // Let integration hide any output that wasn't a fusion output when
-  // `aliasOutputToInput` was called. For example, running mean and var for
-  // batch norm.
-  io_alias_[output] = AliasInfo{
-      .type = type,
-      .aliased_io = input,
-      .hide_output = !output->isFusionOutput()};
+  // AllocationType::Evaluate
+  if (!input) { // Non-aliased outputs marked for expression evaluator.
+    NVF_CHECK(
+        output->isFusionOutput(),
+        "Only fusion outputs can be expression evaluated.");
+    io_alias_[output] =
+        AliasInfo{.type = type, .aliased_io = nullptr, .hide_output = false};
+  }
```

Comment:
The current logic looks like:
```
if (reuse) {
  ...
}
if (alias) {
  change io_alias_
} else {
  change io_alias_
}
```

I suggest the following simplification:
```
if (reuse) {
  everything we need for reuse, e.g., hide_output, alias compatibility check, and change io_alias_. 
  return;  // early exit so the reader has only one type to reason about for the rest of the function 
}

assert type is evaluate;
assert output is a fusion output; 
// As I mentioned in the other comment, you could check isAliasCompatible, but IMHO the benefits of the minimal and maybe fragile check over there is not worth the added code complexity. 
add {type, input, false};
```

Reply:
```
if (!output->isFusionOutput()) {
      addOutput(output);
    }
``` 
What cases does this cover?